examples/addr2line.rs

```diff
@@ -0,0 +1,208 @@
+extern crate gimli;
+extern crate getopts;
+
+use std::env;
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let mut opts = getopts::Options::new();
+    opts.optopt("e", "exe", "Set the input file name (default is a.out)", "<executable>");
+
+    let matches = opts.parse(&args[1..]).unwrap();
+    let file_path = matches.opt_str("e").unwrap_or("a.out".to_string());
+    let file = obj::open(&file_path);
+    if obj::is_little_endian(&file) {
+        symbolicate::<gimli::LittleEndian>(&file, &matches);
+    } else {
+        symbolicate::<gimli::BigEndian>(&file, &matches);
+    }
+}
+
+fn parse_uint_from_hex_string(string: &str) -> u64
+{
+    if string.len() > 2 {
+        if string.starts_with("0x") {
+            return u64::from_str_radix(&string[2..], 16).expect("Failed to parse address");
+        }
+    }
+
+    return u64::from_str_radix(string, 16).expect("Failed to parse address");
+}
+
+fn entries_for_addresses<Endian>(file: &obj::File, addrs: &Vec<u64>) -> Vec<Option<gimli::DebugInfoOffset>>
+    where Endian: gimli::Endianity
+{
+    let aranges = obj::get_section(file, ".debug_aranges").expect("Can't addr2line with no aranges");
+    let aranges = gimli::DebugAranges::<Endian>::new(aranges);
+    let mut aranges = aranges.aranges();
+
+    let mut dies: Vec<Option<gimli::DebugInfoOffset>> = (0..addrs.len()).map(|_| None).collect();
+    while let Some(arange) = aranges.next_arange().expect("Should parse arange OK") {
+        let start = arange.start();
+        let end = start + arange.len();
+
+        for (i, addr) in addrs.iter().enumerate() {
+            if *addr >= start && *addr < end {
+                dies[i] = Some(arange.debug_info_offset());
+            }
+        }
+    }
+
+    dies
+}
+
+fn offset_for_entry<Endian>(abbrevs: &gimli::DebugAbbrev<Endian>, header: &gimli::UnitHeader<Endian>)
+                            -> Option<gimli::DebugLineOffset>
+    where Endian: gimli::Endianity
+{
+    let abbrev = abbrevs.abbreviations(header.debug_abbrev_offset()).expect("Fail");
+    let mut entries = header.entries(&abbrev);
+    let (_, entry) = entries.next_dfs().expect("Fail").unwrap();
```

Comment:
Nit:

```rust
let (_, entry) = entries.next_dfs()
    .expect("Should parse first entry OK")
    .expect("And first entry should exist");
```

The whole point of using `expect("...")` rather than `unwrap()` is to get better error messages when things go wrong, so we should write messages that give us some context or idea of what went wrong if/when things fail.